compiler/rustc_middle/src/lint.rs

```diff
@@ -225,6 +225,7 @@ pub fn explain_lint_level_source(
                 err.note_once(format!(
                     "`{flag} {hyphen_case_lint_name}` implied by `{flag} {hyphen_case_flag_val}`"
                 ));
+                err.help_once(format!("use `#[allow({name})]` to allow"));
```

Comment:
I like how actionable this help message is but I wonder if it would be a good idea to word it slightly differently. Unless you already know what allowing a lint does, it may not be entirely clear what effect this suggestion will have. Perhaps we should say something like `use #[allow({name})] to suppress this lint`? Thoughts? 🙂

Reply:
Sure. I would love to improve it.

> `use #[allow({name})] to suppress this lint`

Definitely better, although the word 'suppress' isn't used currently.

Perhaps it would be easier to understand if we contextualised the message:
`add #[allow({name})] to the expression to override {flag} {hyphen_case_flag_val}`

Which for the example in the PR description would be:
`add #[allow(dead_code)] to the expression to override -W unused`

Reply:
Is it always an expression?

Reply:
> Is it always an expression?

Strictly speaking, [no](https://doc.rust-lang.org/reference/attributes.html), but the word 'expression' is used loosely here to mean that the attribute must be applied to what is being lined.

I realise that is may not be the best word but I don't have another one.

Reply:
I was just thinking about items, i.e. there are common cases like `#[allow(...)] mod a {}`.

> maybe "to the expression" could be removed, ie just add `#[allow(dead_code)] to override -W unused` ?

Sounds good to me, and it is less verbose, which is always good.